cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py

```diff
@@ -104,6 +104,11 @@ def install_kubernetes_components():
         hookenv.log(install)
         check_call(install)
 
+    # Used by the "registry" action. The action is run on a single worker, but
+    # the registry pod can end up on any worker, so we need this directory on
+    # all the workers.
+    os.makedirs('/srv/registry', exist_ok=True)
```

Comment:
This is a prickly solution though. Deeply tying the registry's affinity to a single unit seems like we're headed for  "We gave it some raw disk on one of the workers, it got rescheduled and we lost all the images until we can sync those over... but now the disk is full on worker2 because it only had 10GB to begin with." 

It really seems like we should force the registry to bind to a PersistentVolumeClaim and leave the Persistent Volume empty. We could handle this one of two ways.

For default out of the box setup - enlist and mount an EmptyDir volume. This by design will nuke data when the workload gets rescheduled. It's less desirable but would work for demo purposes.

A storage admin can then come by and enlist another PersistentVolume backed by NFS, CIFS, CloudStorage, Ceph, etc. and Give it a proper floating storage backend, presumably connected to a SAN somewhere. 

Just food for thought to make it more useful out of the box in more places than tying it to a single node without any affinity in the nodeselector.